### ISSUE TITLE
Deploy reload doesn't seem to work - IOS

### ISSUE THREAD

**Issue by sagivf** [driftyco/ionic-cloud#83]
http://docs.ionic.io/services/deploy/#reload-the-app

```
Cordova CLI: 6.4.0
Gulp version:  CLI version 3.9.0
Gulp local:   Local version 3.9.1
Ionic Framework Version: 2.0.0-rc.1
Ionic CLI Version: 2.1.4
Ionic App Lib Version: 2.1.2
Ionic App Scripts Version: 0.0.36
ios-deploy version: 1.8.3
ios-sim version: 5.0.4
OS: Mac OS X Sierra
Node Version: v6.9.1
Xcode version: Xcode 8.0 Build version 8A218a
````

My code:

``` javascript
  doUpdate() {
    this.checkUpdateStatus = 'Updating...';
    this.deploy.download().then(() => {
      return this.deploy.extract();
    }).then(() => {
      this.checkUpdateStatus = 'Update Complete!';
      return this.deploy.load();
    }).catch(() => {
      this.checkUpdateStatus = 'Update Failed';
    });
  }
```

**Comment by sagivf** [comment]
:worried:

**Comment by durasj** [comment]
Hi @sagivf,

Can you please confirm that on .load method call cordova.js is not found? I have such problem with everything up-to-date and the App needs to be completely reloaded on the native level (which AFAIK can't be done in the code), location.reload won't work.

Thanks

**Comment by sagivf** [comment]
@durasj sorry for the delayed response.
.load is `typeof function` so that's ok.

Not sure what's going on. It either doesn't reload. Or reloads without the new version.
Currently the app user needs to restart the app.

**Comment by durasj** [comment]
@sagivf it was probably not clear enough what I was trying to say, sorry.

I meant that after the .load function is called, error message "cordova.js is not found" appears in my case. After any number of reloads via web api, the new version won't be loaded. I wanted to know whether it's same for you so it could potentially help guys on Ionic team.

**Comment by sagivf** [comment]
@durasj no problem 👍 
I'm not getting that error. Not sure I understand what you mean by `reload via web api` do any other Ionic cloud services work for you? Are you able to use `this.deploy.check()` for instance?

**Comment by durasj** [comment]
@sagivf sorry for not being more clear. I was talking about traditional Web API [Location.reload()](https://developer.mozilla.org/en-US/docs/Web/API/Location/reload) which I also tried after using Deploy.load(). Other services are fully working for me.

**Comment by sagivf** [comment]
Cool, got ya. I'll try to check again but I don't think I'm getting that error though we seem to be having the same issue. Ionic Team?

**Comment by dwieeb** [comment]
@sagivf Does `checkUpdateStatus` actually change to `"Update Failed"`?

**Comment by sagivf** [comment]
@dwieeb No, It just doesn't do anything. I think it might have reloaded once, but the update didn't apply. Just to be clear - manually restarting the app by the user gets the update applied.

**Comment by durasj** [comment]
@dwieeb same here, it looks like the update succeeded, but I also have the cordova.js not found right after calling .load() or manually using Location.reload().

**Comment by andkojanci** [comment]
Hello guys,

I've exactly same issue. We need to release the application to production but our updating mechanism still cannot reload the application.

Ionic team, please help!

**Comment by sagivf** [comment]
:worried:

**Comment by Fuiste** [comment]
Hey everyone.  I've tried unsuccessfully to reproduce this issue locally on an iPhone, and the service appears to be running as intended on my end.

When looking at the native xcode logs, are there any errors or anything unusual jumping out at you?

**Comment by sagivf** [comment]
@Fuiste I'll try to check it out, but I'm not sure it's relevant if there is no solution for this - #111

**Comment by durasj** [comment]
Hi @Fuiste. In my case, it looks like [#99 @ ionic-plugin-deploy](https://github.com/driftyco/ionic-plugin-deploy/issues/99) is the possible problem - wkwebview. So I guess it should be easily reproducible using wkwebview, but I guess ionic-cloud client is therefore not responsible.

**Comment by sagivf** [comment]
@Fuiste  @durasj I am also using wkwebview, so theres a good chance that is the issue here.

**Comment by Fuiste** [comment]
@sagivf and @durasj yup, it's an issue with WkWebView that we were recently able to identify.  We're testing a couple different solutions at present, and hope to have a new release of the plugin as soon as possible.

**Comment by ciesielskico** [comment]
Any news on reloading WKWebview? Running into the same issue with Ionic Deploy.

**Comment by timbru31** [comment]
See https://github.com/ionic-team/ionic-plugin-deploy/issues/99#issuecomment-304952231 for a possible solution (uses fork of cordova-plugin-wkwebview-engine)

**Comment by durasj** [comment]
Hi @Fuiste, could you please confirm that one of the solutions that you mentioned you're preparing is in the ionic-plugin-deploy changes? If so, do you please plan to include it in normal tagged release?
